src/nvim/os/env.c

```diff
@@ -2,10 +2,20 @@
 
 #include <uv.h>
 
+// vim.h must be included before charset.h (and possibly others) or things
+// blow up
```

Comment:
Declarations generator is completely irrelevant. All it does is putting declarations in two files. Including correct headers that do necessary typedefs is the job for the developer.

Only non-generated headers are intended to “just work”, but even this is not the case with headers like `buffer.h`.